index.js

```diff
@@ -137,14 +366,31 @@ module.exports = (opts = {}) => {
     ],
     opts.unwrap
   )
+  let rootRuleName = (opts.rootRuleName || 'at-root').replace(/^@/, '')
   let preserveEmpty = opts.preserveEmpty
 
+  let hasRootRules = false
```

Comment:
Looking at this again, I'm pretty sure this will not work, if the plugin is instantiated once and then used for multiple CSS inputs.

@ai do you know if there's a way to store local state in a plugin, so that I can avoid redundant work, like in this case?

Reply:
ok, found the solution.

[99823d2](https://github.com/postcss/postcss-nested/pull/147/commits/99823d20f4347367db58551d63153111054b488c)